pkg/hubble/monitor/consumer.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"time"
+
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	monitorConsumer "github.com/cilium/cilium/pkg/monitor/agent/consumer"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Observer is the receiver of MonitorEvents
+type Observer interface {
```

Comment:
I think it would make sense to compose the `GRPCServer` interface in `pkg/hubble/observer`, which also defines these 2 methods, with this one.

Reply:
Yes, I might do that. But it can only done in  #12712 (GetEventsChannel still has the wrong signature on this branch)